plumbing/format/idxfile/idxfile.go

```diff
@@ -57,9 +57,9 @@ type MemoryIndex struct {
 	PackfileChecksum plumbing.Hash
 	IdxChecksum      plumbing.Hash
 
-	offsetHash       map[int64]plumbing.Hash
-	offsetHashIsFull bool
```

Comment:
I would convert this to an atomic and keep it because current approach sacrifices performance in  `FindOffset` by keep writing even after full generation. Combining `buildOnce` with atomic would provide better performance